src/core/annotation.js

```diff
@@ -237,32 +303,6 @@ var Annotation = (function AnnotationClosure() {
       }
     },
 
-    isInvisible: function Annotation_isInvisible() {
-      var data = this.data;
-      return !!(data &&
-                data.annotationFlags &&            // Default: not invisible
-                data.annotationFlags & 0x1);       // Invisible
-    },
-
-    isViewable: function Annotation_isViewable() {
-      var data = this.data;
-      return !!(!this.isInvisible() &&
-                data &&
-                (!data.annotationFlags ||
-                 !(data.annotationFlags & 0x22)) &&  // Hidden or NoView
-                data.rect);                          // rectangle is necessary
-    },
-
-    isPrintable: function Annotation_isPrintable() {
-      var data = this.data;
-      return !!(!this.isInvisible() &&
```

Comment:
The spec is not really clear about if this, but looking at https://github.com/mozilla/pdf.js/blob/99f9b523c2c4092d370934612338956f566edf2d/src/core/annotation.js#L332-L333 I think you are right. I think the correct way to do this is to check `isViewable` too when `isPrintable` is called, because it does not make sense to print something that is not viewable in the first place. This might also address the regression you mention. I'll address this soon.